packages/sdks/src/functions/get-fetch.ts

```diff
@@ -1,14 +1,36 @@
 import { getGlobalThis } from './get-global-this.js';
 
+/**
+ * `node-fetch` is a polyfill for fetch.
+ *
+ * There are two environments:
+ * 1. Node - in which case we may need to fetch `node-fetch` and use it.
+ * 2. Native (browser/deno/edge-workers) - in which case we can use the native
+ *   `fetch` function.
+ *
+ * PROBLEM:
+ * If we write `import('node-fetch')` the bundler will always bundle `node-fetch`
+ * because it does not know if it is needed. This is not what we want, because:
+ * - in native environments will now get `fetch` and the `node-fetch` polyfill is
+ *   not needed.
+ * - in Node environment bundling the `node-fetch` polyfill provides no benefit as
+ *   we can just `require('node-fetch')` it.
+ *
+ * ENSURE: that the bundler does not bundle `node-fetch`.
+ */
+
+let fetch: typeof global.fetch | undefined = undefined;
+
 export async function getFetch(): Promise<typeof global.fetch> {
+  if (fetch) return fetch;
   const globalFetch: typeof global.fetch = getGlobalThis().fetch;
 
   if (typeof globalFetch === 'undefined' && typeof global !== 'undefined') {
-    const nodeFetch = import('node-fetch').then((d) => d.default) as Promise<
+    const nodeFetch = import(`node-fetch`).then((d) => d.default) as Promise<
       typeof global.fetch
     >;
-    return (nodeFetch as any).default || nodeFetch;
+    return (fetch = (nodeFetch as any).default || nodeFetch);
   }
 
-  return (globalFetch as any).default || globalFetch;
+  return (fetch = (globalFetch as any).default || globalFetch);
```

Comment:
can we undo all the changes to this file now?

Reply:
done